File: TaskTrackerWebAPI/src/Services/Dtos/TaskDto.cs
using System;
using TaskTrackerWebAPI.src.Data;

namespace TaskTrackerWebAPI.src.Services.Dtos;

public record TaskDto
{

    public int? Id { get; set; }
    public string? Title { get; set; } = string.Empty;
    public string? Description { get; set; } = string.Empty;
    public string? Status { get; set; } = string.Empty;
    public string? Priority { get; set; } = string.Empty;
    public DateTime? DueDate { get; set; }
    public DateTime? CreatedAt { get; set; }

}

File: TaskTrackerFrontend/src/app/models/task_dto.ts
export interface TaskDTO {
    id: number;
    title: string;
    description: string;
    status: string;
    priority: string;
    dueDate: string;
    createdAt: string;
}

export interface CreateTaskDTO {
    title: string;
    description: string;
    status: string;
    priority: string;
    dueDate: string;
}

export interface PaginatedResponseDTO<T> {
    data: T[];
    total: number;
    page: number;
    pageSize: number;
}